src/resources/lib/modules/system.py

```diff
@@ -578,16 +660,42 @@ def do_manual_update(self, listItem=None):
         except Exception, e:
             self.oe.dbg_log('system::do_manual_update', 'ERROR: (' + repr(e) + ')')
 
-    def get_json(self):
+    def get_json(self, url=None):
         try:
             self.oe.dbg_log('system::get_json', 'enter_function', 0)
-            url = self.UPDATE_DOWNLOAD_URL % ('releases', 'releases.json')
-            update_json = json.loads(self.oe.load_url(url))
-            return update_json
+            if url is None:
+                url = self.UPDATE_DOWNLOAD_URL % ('releases', 'releases.json')
+            if url.split('/')[-1] != 'releases.json':
+                url = url + '/releases.json'
+            data = self.oe.load_url(url)
+            if not data is None:
+                update_json = json.loads(self.oe.load_url(url))
```

Comment:
This is loading the url a second time if it worked the first time, doubling the network overhead... shouldn't this be:
```
+                update_json = json.loads(data)
```
?